app/src/main/java/sw19/moring03/paint/MainActivity.java

```diff
@@ -99,10 +98,14 @@ public boolean onPrepareOptionsMenu(Menu menu) {
         setToolIcon();
         setStrokeWidth(strokeWidth);
         DrawingView view = findViewById(R.id.drawingView);
+        MenuItem backButton = menu.findItem(R.id.undoButton);
+
         if (view.isAlreadyDrawn()) {
-            menu.findItem(R.id.undoButton).setVisible(true);
+            backButton.setEnabled(true);
+            backButton.getIcon().setAlpha(255);
         } else {
-            menu.findItem(R.id.undoButton).setVisible(false);
+            backButton.setEnabled(false);
+            backButton.getIcon().setAlpha(40);
```

Comment:
When undo button is only grayed out and the line gets selected as drawing tool the line type button appears to the left of the grayed out undo button. Instead this button should be appear on the right side of the button.